src/opdsrequestmanager.cpp

```diff
@@ -21,6 +26,44 @@ int OpdsRequestManager::getCatalogPort()
          : 443;
 }
 
+// Helper to handle replies and follow redirects
+void OpdsRequestManager::handleReply(QNetworkReply* reply, std::function<void(QNetworkReply*)> finalHandler, int redirectCount) {
+    QVariant redirectAttr = reply->attribute(QNetworkRequest::RedirectionTargetAttribute);
+    if (redirectAttr.isValid() && redirectCount < kMaxRedirects) {
+        QUrl redirectUrl = redirectAttr.toUrl();
+        // Make absolute if needed
+        if (redirectUrl.isRelative()) {
+            redirectUrl = reply->url().resolved(redirectUrl);
+        }
+        qInfo() << "Following redirect" << redirectCount + 1 << "to:" << redirectUrl.toString();
+        reply->deleteLater();
+        QNetworkRequest newRequest(redirectUrl);
+        QNetworkReply* newReply = m_networkManager.get(newRequest);
+        connect(newReply, &QNetworkReply::finished, this, [=]() {
+            handleReply(newReply, finalHandler, redirectCount + 1);
+        });
+        return;
+    }
+    // No redirect, or max redirects reached
+    if (redirectAttr.isValid() && redirectCount >= kMaxRedirects) {
```

Comment:
With current code the `redirectCount >= kMaxRedirects` check is redundant. However simply deleting it will lead to confusion. Please merge this `if` statement with the previous one and check the redirect count in a nested `if/else` statement.